signature/mechanism_openpgp.go

```diff
@@ -177,3 +178,101 @@ func (m *openpgpSigningMechanism) Verify(unverifiedSignature []byte) (contents [
 func (m *openpgpSigningMechanism) UntrustedSignatureContents(untrustedSignature []byte) (untrustedContents []byte, shortKeyIdentifier string, err error) {
 	return gpgUntrustedSignatureContents(untrustedSignature)
 }
+
+// isSubkeyOf checks if signerKeyIdentity is a subkey of expectedKeyIdentity
+func (m *openpgpSigningMechanism) isSubkeyOf(signerKeyIdentity, expectedKeyIdentity string) (bool, error) {
+	// Convert fingerprints to lowercase for comparison
+	signerFingerprint := strings.ToLower(signerKeyIdentity)
+	expectedFingerprint := strings.ToLower(expectedKeyIdentity)
+
+	// If they're the same, it's a match
+	if signerFingerprint == expectedFingerprint {
+		return true, nil
+	}
+
+	// Find the entity with the expected fingerprint
+	var expectedEntity *openpgp.Entity
+	var signerEntity *openpgp.Entity
+
+	for _, entity := range m.keyring {
+		// Check if this entity's primary key matches the expected fingerprint
+		if entity.PrimaryKey != nil {
+			primaryFingerprint := strings.ToLower(fmt.Sprintf("%x", entity.PrimaryKey.Fingerprint))
+			if primaryFingerprint == expectedFingerprint {
+				expectedEntity = entity
+			}
+			if primaryFingerprint == signerFingerprint {
+				signerEntity = entity
+			}
+		}
+
+		// Also check subkeys
+		for _, subkey := range entity.Subkeys {
+			if subkey.PublicKey != nil {
+				subkeyFingerprint := strings.ToLower(fmt.Sprintf("%x", subkey.PublicKey.Fingerprint))
+				if subkeyFingerprint == expectedFingerprint {
+					expectedEntity = entity
+				}
+				if subkeyFingerprint == signerFingerprint {
+					signerEntity = entity
+				}
+			}
+		}
+	}
+
+	// If we couldn't find either key, return false
+	if expectedEntity == nil || signerEntity == nil {
+		return false, nil
+	}
+
+	// Case 1: Both keys belong to the same entity - this is valid
+	if expectedEntity == signerEntity {
+		return true, nil
+	}
+
+	// Case 2: The signer is a subkey of the expected entity
+	if expectedEntity.PrimaryKey != nil {
+		expectedPrimaryFingerprint := strings.ToLower(fmt.Sprintf("%x", expectedEntity.PrimaryKey.Fingerprint))
+		if expectedPrimaryFingerprint == expectedFingerprint {
+			// The expected key is a primary key, check if signer is one of its subkeys
+			for _, subkey := range expectedEntity.Subkeys {
+				if subkey.PublicKey != nil {
+					subkeyFingerprint := strings.ToLower(fmt.Sprintf("%x", subkey.PublicKey.Fingerprint))
+					if subkeyFingerprint == signerFingerprint {
+						return true, nil
+					}
+				}
+			}
+		}
+	}
+
+	// Case 3: The expected key is a subkey, and the signer is the primary key of the same entity
```

Comment:
Similarly to the GPG case, with `expectedFingerprint` matching a subkey, ignoring that and accepting any other subkey is surprising.